src/vs/workbench/electron-browser/main.contribution.ts

```diff
@@ -217,6 +217,11 @@ configurationRegistry.registerConfiguration({
 			'description': nls.localize('revealIfOpen', "Controls if an editor is revealed in any of the visible groups if opened. If disabled, an editor will prefer to open in the currently active editor group. If enabled, an already opened editor will be revealed instead of opened again in the currently active editor group. Note that there are some cases where this setting is ignored, e.g. when forcing an editor to open in a specific group or to the side of the currently active group."),
 			'default': false
 		},
+		'workbench.editor.collapseEmptyGroupsOnClose': {
```

Comment:
The setting name is a bit hard to understand. How about `removeEmptyGroups` or `collapseEmptyGroups` and then explaining in the description that it applies when you close the last editor?